src/api/janet.c

```diff
@@ -0,0 +1,140 @@
+#include "api.h"
+#include "core/core.h"
+#include <assert.h>
+
+#if defined(TIC_BUILD_WITH_JANET)
+
+#include "janet.h"
+
+static const char *const JanetKeywords[] = {
+    // special forms
+    "def", "var", "fn", "do", "quote", "if", "splice", "while", "break", "set",
+    "quasiquote", "unquote", "upscope",
+    // literals
+    "nil", "true", "false",
+    // common macros
+    // TODO: decide which ones should be hightlighted as keywords
+    // "%=","*=","++","+=","--","-=","->","->>","-?>","-?>>","/=",
+    // "and","as->","as-macro","as?->","assert","case","chr","comment","compif","comptime","compwhen","cond","coro","def-","default","defdyn","defer","defmacro","defmacro-","defn","defn-","delay","doc","each","eachk","eachp","edefer",
+    // "ev/do-thread","ev/gather","ev/spawn","ev/spawn-thread","ev/with-deadline","ffi/defbind",
+    // "fiber-fn","for","forever","forv","generate","if-let","if-not","if-with","import","juxt","label","let","loop","match","or","prompt","protect","repeat","seq","short-fn","tabseq","toggle","tracev","try","unless","use","var-","varfn","when","when-let","when-with","with","with-dyns","with-syms","with-vars",
+};
+
+static const char* DYN_TIC_MEM = ":tic_mem";
+
+#define JANET_FN_S_tic(CNAME) \
+    static const int32_t _cfun_##CNAME##_sourceline_ = __LINE__; \
+    static Janet _cfun_##CNAME (int32_t argc, Janet *argv) { \
+        tic_mem* tic = janet_unwrap_pointer(janet_dyn(DYN_TIC_MEM)); \
+        assert(tic != NULL);
+
+JANET_FN_S_tic(btn)
+    janet_fixarity(argc, 1);
```

Comment:
Just to be clear this param is optional, not required.

Reply:
that's not in the doc

Reply:
I would agree it's unclear.

`btn([id]) -> pressed`

The `[]` means the argument is optional.  If no specific button is passed I believe the entire state of the button register (bitfield) is returned. (you can read the source to confirm)

I'm speaking largely of Lua, it might be useful to check and see if this has been handled correctly across all runtimes.

Reply:
Yes, `btn` and `btnp` have the same behavior for 0 arguments:

https://github.com/nesbox/TIC-80/blob/main/src/api/lua.c#L601

Reply:
it seem to always return `true` for me in the Lua version.